### ISSUE TITLE
Unable to click suggestions on mobile Safari

### ISSUE THREAD

**Issue by jznadams** [devbridge/jQuery-Autocomplete#542]
After integrating this into my rails application, I discovered that it worked great in every browser except for mobile Safari-based browsers on iOS devices. In that case I was unable to click on any of the suggestions. It seemed like the onClick event was never being transmitted to the JavaScript. (FYI it works fine in the iOS simulator, the problem only appeared for me on actual iOS devices)

After a lot of trial and error, I stumbled upon [this explanation](http://www.shdon.com/blog/2013/06/07/why-your-click-events-don-t-work-on-mobile-safari) (the second case). It seems that mobile Safari has a weird requirement that click events are only triggered if the CSS of the object being tapped has the cursor set to pointer.  I added the following to my CSS and everything started working:

```
.autocomplete-suggestion { 
  cursor: pointer;
}
```

Maybe it would be worth adding that CSS line to the example styles in the readme? It certainly would have saved me a bunch of debug time.

Thanks for a great library!

**Comment by jznadams** [comment]
I also ended up noticing that scrolling inside of the search results was not working properly on mobile Safari. An additional set of CSS styles fix that as well:

```
.autocomplete-suggestions { 
  overflow: auto;
  -webkit-overflow-scrolling: touch;
  overflow-y: scroll;
}
```

Here are some resources on the specific weirdness for Safari for those that are interested:
[Apple Developer Docs](https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html#//apple_ref/doc/uid/TP40006511-SW6)
[Mozilla Foundation](https://developer.mozilla.org/en-US/docs/Web/Events/click#Safari_Mobile)

Also for the sake of other people that run into this, I ran into an additional problem where the onClick event in the suggestions would _still_ not be fired if the size of the webpage suggestions appear on is too long. In my particular case I had a search bar at the top of a very long bootstrap responsive table with around 3000 rows. If there were only 30 rows shown, everything works fine.

The only workaround I could come up with is to use JavaScript to remove the table from the DOM at the beginning of search (`beforeRender:`) and then put it back when the suggestions container hides (`onHide:`).  Just hiding the table didn't seem to work, I had to physically remove it (i.e. `elem.detach()`) so that the total size of the page shrank. So weird.
